src/mui-themeable.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import DefaultRawTheme from './styles/raw-themes/light-raw-theme';
+import ThemeManager from './styles/theme-manager';
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+export default function muiThemeable(WrappedComponent) {
+  const MuiComponent = (props, {muiTheme = ThemeManager.getMuiTheme(DefaultRawTheme)}) => {
+    return <WrappedComponent {...props} muiTheme={muiTheme} />;
+  };
+
+  MuiComponent.displayName = `mui(${getDisplayName(WrappedComponent)})`;
```

Comment:
Yeah I'd have to look at how other components are using `displayName` currently. I'm not sure if this makes a difference, but this component isn't changing the name of the component, it's just changing the name of the high order component to be `mui(X)`. So the X component still lives in the tree underneath this and is appropriately named `X`.

I haven't tested this approach beyond the Divider component we merged recently (which is a very simple component). Later in a this branch I'll convert some of the more complex components and those that use displayName to use muiThemeable and investigate the impact.